### PR TITLE
LINK-1236 | Require authentication before enrolment

### PR DESCRIPTION
**Pull request by jorilindell**
## Description
- Allow only authenticated users to fill registration form or edit his own signup
- Show notification which advices user to log in if user is unauthenticated

## Closes :no_good_woman:
[LINK-1236](https://helsinkisolutionoffice.atlassian.net/browse/LINK-1236)

## Testing :alembic:
Show notification is unauthenticated user opens create enrolment page
Screenshot 2023-08-22 at 17 17 32

Show log in page is unauthenticated user opens edit enrolment page
Screenshot 2023-08-22 at 17 17 41

[LINK-1236]: https://helsinkisolutionoffice.atlassian.net/browse/LINK-1236?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/domain/enrolment/__tests__/utils.test.ts [hunk 1]
```diff
@@ -90,8 +85,9 @@ describe('getEnrolmentPayload function', () => {
       dateOfBirth = '10.10.1999',
       email = 'Email',
       extraInfo = 'Extra info',
+      firstName = 'First name',
+      lastName = 'First name',
```
Comment (mikkojamG) [inline src/domain/enrolment/__tests__/utils.test.ts:89]: Should probably be lastName = 'Last name',

**Review by mikkojamG — APPROVED** [review]
One minor comment, otherwise LGTM 👍

File: src/domain/enrolment/__tests__/utils.test.ts (hunk 1 above)
Comment (jorilindell) [reply src/domain/enrolment/__tests__/utils.test.ts:89]: Oh, sure it should. Good catch.

Comment (jorilindell) [reply src/domain/enrolment/__tests__/utils.test.ts]: Fixed in https://github.com/City-of-Helsinki/linkedregistrations-ui/pull/60/commits/4e199294470841d168d96c5821740600f23d5555

**Pull request merged**